src/evm/process-transaction.ts

```diff
@@ -0,0 +1,75 @@
+import * as node from '@api3/airnode-node';
+import { ethers } from 'ethers';
+import { GAS_LIMIT } from '../constants';
+import { ProcessableSubscription } from '../types';
+import { retryGo } from '../utils';
+
+export const processTransaction = async (
+  airnodeWallet: ethers.Wallet,
+  contract: ethers.Contract,
+  gasTarget: node.GasTarget,
+  subscriptions: ProcessableSubscription[],
+  sponsorWallet: ethers.Wallet
+): Promise<node.LogsData<ProcessableSubscription>[]> => {
+  const logs: node.LogsData<ProcessableSubscription>[] = [];
+
+  // Process each subscription in serial to keep nonces in order
+  for (const subscription of subscriptions.sort((a, b) => a.nonce - b.nonce) || []) {
```

Comment:
I feel like I read this exact comment somewhere else:
Are you sure these need to be done serially?
While working on another task recently I routinely did 7 transactions asynchronously with the nonces incremented in a `.map` and I never received any errors. You should be able to convert this for loop to a map and the return from the map via `Promise.all` could be flattened into `logs`, which would remove the type and initialisation. Lmk if you disagree, once again I'm curious.

Reply:
> I feel like I read this exact comment somewhere else:

probably [here](https://github.com/api3dao/airkeeper/blob/bec210-refactor-airkeeper/src/handlers/rrp.ts#L209) 😁 

> Are you sure these need to be done serially?

Not 💯  sure but it when writing the RRP beacon update my concern was that if a tx gets rejected or something then the nonce order would get messed up.

> While working on another task recently I routinely did 7 transactions asynchronously with the nonces incremented in a .map and I never received any errors. You should be able to convert this for loop to a map and the return from the map via Promise.all could be flattened into logs, which would remove the type and initialisation. Lmk if you disagree, once again I'm curious.

I will definitely test this and if it works then I would prefer to run them all in parallel.

Reply:
Is there a scenario where a middle transaction would fail but the preceding tx and following txs would/would be able to succeed? 

This is assuming the chain services/permissions have been correctly configured.

Reply:
🤔 not sure to be honest, maybe a timeout from a provider or something totally unexpected like that.

The thing is that we probably don't want to keep submitting txs if for instance sponsor wallet runs out of funds. My thinking was that as soon as a tx reverts or fails then we stop sending txs with this wallet.